src/content/en/updates/2017/12/modulepreload.md

```diff
@@ -0,0 +1,138 @@
+way of letting the browser know straight away about critical files that will be needed
+as part of the current navigation. When the browser sees the preload, it starts a high
+priority download for the resource, so that by the time it's actually needed it's either
+already fetched or partly there.
+
+## OK, so why doesn't `<link rel="preload">` work for modules?
+
+This is where things get tricky. There are several credentials modes for
+resources, and in order to get a cache hit they must match, otherwise you end up
+fetching the resource twice. Needless to say, double-fetching is bad, because it
+wastes the user's bandwidth and makes them wait longer, for no good reason.
+
+For `<script>` and `<link>` tags, you can set the credentials mode with the `crossorigin`
+attribute. However, it turns out while that a `<script type="module">` with no
+`crossorigin` attribute means that the credentials mode is `omit`, which doesn't exist
+for `<link rel="preload">`. This means that you would have to
+change the `crossorigin` attribute in both your `<script>` and `<link>` to one
+of the other values, and you might not have an easy way of doing so if what you're
+trying to preload is a dependency of other modules.
+
+Furthermore, fetching the file is only the first step in actually running the code;
+it first has to be parsed and compiled by Chrome's JavaScript engine - V8. Ideally,
+this should happen ahead of time as well, so that when the module is needed, the code is
+ready to run. However, V8 parses and compiles modules differently from other JavaScript,
+and needs to know which of the two it's looking at. `<link rel="preload">` doesn't
+provide any way of indicating that the file being loaded is a module, so all the browser
+can do is load the file and put it in the cache. Only when a request for the module is
+made from a `<script type="module">` tag or another module will it be able to parse and
+compile the code as a module.
+
+## So is `<link rel="modulepreload">` just `<link rel="preload">` for modules?
+
+In a nutshell, yes. By having a specific `link` type for preloading modules, we can
+write simple HTML without worrying about what credentials mode we're using.
+
+```html
+<head>
+  <link rel="modulepreload" href="super-critical-stuff.js">
+</head>
+[...]
+<script type="module" src="super-critical-stuff.js">
+```
+And since Chrome now knows that what you're preloading is a module, it can be smart
+and parse and compile the module as soon as it's done fetching, instead of waiting
+until it tries to run.
+
+## But what about modules' dependencies?
+
+Funny you should ask! There is indeed something we haven't talked about: recursivity.
+
+The `<link rel="modulepreload">` spec actually allows for optionally loading not just
+the requested module, but all of its dependency tree as well. Browsers don't have to
+do this, but they can.
+
+So what would be the best practice for preloading a module and its dependency tree
```

Comment:
be the best practice for preloading -> be the best cross browser solution for preloading

and remove the last part of the sentence.

Reply:
Done.